test/common/config/delta_subscription_test_harness.h

```diff
@@ -88,10 +89,13 @@ class DeltaSubscriptionTestHarness : public SubscriptionTestHarness {
       error_detail->set_message(error_message);
     }
     EXPECT_CALL(async_stream_,
-                sendMessage(ProtoEqIgnoringField(expected_request, "response_nonce"), false))
-        .WillOnce([this](const Protobuf::Message& message, bool) {
-          const std::string nonce =
-              static_cast<const envoy::api::v2::DeltaDiscoveryRequest&>(message).response_nonce();
+                sendMessageRaw_(
+                    Grpc::ProtoBufferEqIgnoringField(expected_request, "response_nonce"), false))
+        .WillOnce([this](Buffer::InstancePtr& buffer, bool) {
+          Buffer::ZeroCopyInputStreamImpl stream(std::move(buffer));
+          envoy::api::v2::DeltaDiscoveryRequest message;
+          EXPECT_TRUE(message.ParseFromZeroCopyStream(&stream));
+          const std::string nonce = message.response_nonce();
```

Comment:
This pattern comes up a few times, can you refactor?

Reply:
done.